### PR TITLE
gadget: remove partition table data from ondisk volume

### PR DESCRIPTION
**Pull request by cmatsuoka**
The initial method for creating partitions during installation required
a modified partition table to be sent to the partitioning utility. This
was changed to just inform about the newly created partitions, so keeping
the entire partition table in OnDiskVolume is no longer necessary.

This is the first step of the OnDiskVolume/LaidOutVolume unification
and aims to define OnDiskVolume and OnDiskStructure as its LaidOut
counterparts plus extra fields (only the device node name and a flag
for partitions created during install are needed).

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

### CODE REVIEW

File: gadget/ondisk.go [hunk 1]
```diff
@@ -131,15 +131,10 @@ type OnDiskStructure struct {
 // OnDiskVolume holds information about the disk device including its partitioning
 // schema, the partition table, and the structure layout it contains.
 type OnDiskVolume struct {
+	LaidOutVolume
```
Comment (pedronis) [inline gadget/ondisk.go:134]: now we have double structures, not sure it's a win. This should probably just drop the the unexported field for now.

Comment (cmatsuoka) [reply gadget/ondisk.go:134]: This was intended to be an intermediate step to be followed by replacing OnDiskVolume with LaidOutVolume with the extra needed fields added, but we can go back and only drop the partition table as the first step.

Comment (cmatsuoka) [reply gadget/ondisk.go]: Done.

**Comment by anonymouse64** [comment]
I'll pick up this PR sometime this week I hope

**Comment by pedronis** [comment]
I'm re-applying what this does in one of my own PRs.

**Pull request closed without merging**